### PR TITLE
fix: preview back button behavior

### LINKED ISSUE #22299: [Preview] Edit view : The Back button redirects to the Preview
### Required System information

- Edition: Community
- OS: Mac and WSL
- Environment: Local

### Describe the bug

When opening the preview of an entry, then closing the preview and clicking on the back button, I’m redirected to the preview

### Steps to reproduce the behavior

1. Go to the list view and go to an entry
2. Open the preview, then close it
3. Click on the back button and you will be redirected to the preview

### Expected behavior

It should redirect to the list view (or at least the previous page in the CMS)

### Screenshots

https://github.com/user-attachments/assets/a407821b-4717-43b7-bc6d-568da993863c

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

Updates the preview back button so it uses navigate(-1) from React Router, simulating a click on the browser's back button. It's needed because otherwise, if you go to preview, click back, and click and the edit view's back button, you're take to the preview page again.

I also fixed an issue in the shared BackButton component where the href of the link was the wrong one, it would show the URL of the current page instead of the previous one.

### How to test it?

I think it's better to let you try it and see if it does what you expect. Just avoid starting a session (or reloading) directly on the preview page, because that will rely on the fallback URL, in which case the back/previous logic cannot be managed.

FYI I noticed a bug where if you toggle the draft and publish tabs multiple times on the preview page, clicking on the back button will not do the navigation on the first try. But I'm pretty sure this is a React Router bug because of us embedding the admin inside itself. Try changing the preview URL in the config to any other URL and I think the bug should disappear.

### Related issue(s)/PR(s)

fixes #22299

### CODE REVIEW

File: packages/core/admin/admin/src/features/BackButton.tsx [hunk 1]
```diff
@@ -221,7 +241,8 @@ const BackButton = React.forwardRef<HTMLAnchorElement, BackButtonProps>(
     };
 
     // The link destination from the history. Undefined if there is only 1 location in the history.
-    const historyTo = canGoBack ? history.at(-1) : undefined;
+    // const historyTo = canGoBack ? history.at(-2) : undefined;
```
Comment (markkaylor) [inline packages/core/admin/admin/src/features/BackButton.tsx:244]: Did you wwant to get rid of this?

File: packages/core/admin/admin/src/features/BackButton.tsx [hunk 2]
```diff
@@ -221,7 +241,8 @@ const BackButton = React.forwardRef<HTMLAnchorElement, BackButtonProps>(
...
+    const historyTo = canGoBack ? history.at(currentLocationIndex - 2) : undefined;
```
Comment (remidej) [inline packages/core/admin/admin/src/features/BackButton.tsx:245]:
```suggestion
    const historyTo = canGoBack ? history.at(currentLocationIndex - 2) : undefined;
```

**Review by markkaylor — APPROVED** [review]
Just tested it out.  Looks good to me.

**Comment by simotae14** [comment]
I tested it locally and it works, I have the same error when I toggle draft and publish tabs multiple times, and then when I click on the Close button I am redirected to this page 
Screenshot 2024-12-04 alle 11 35 29

**Comment by remidej** [comment]
@simotae14 can you tell me if you get the same issue if you change the preview URL in the getstarted's config/admin.js to any external domain?

**Comment by simotae14** [comment]
@remidej in case I change the preview URL with an external url I need to click multiple times to the close button before being redirected to the edit view page

**Comment by remidej** [comment]
@simotae14 interesting, can you show the config you set up and share a video?

**Comment by simotae14** [comment]
This is my admin file
`module.exports = ({ env }) => ({
  // autoOpen: false,
  auth: {
    secret: env('ADMIN_JWT_SECRET', 'example-token'),
  },
  apiToken: {
    salt: env('API_TOKEN_SALT', 'example-salt'),
  },
  auditLogs: {
    enabled: env.bool('AUDIT_LOGS_ENABLED', true),
  },
  transfer: {
    token: {
      salt: env('TRANSFER_TOKEN_SALT', 'example-salt'),
    },
  },
  flags: {
    nps: env.bool('FLAG_NPS', true),
    promoteEE: env.bool('FLAG_PROMOTE_EE', true),
  },
  preview: {
    enabled: env.bool('PREVIEW_ENABLED', true),
    config: {
      handler: (uid, { documentId, locale, status }) => {
        const kind =
          strapi.contentType(uid).kind === 'collectionType' ? 'collection-types' : 'single-types';
        // return `/admin/preview/${kind}/${uid}/${documentId}/${locale}/${status}`;
        return 'http://localhost:3000/en';
      },
    },
  },
});
`

and here a video


https://github.com/user-attachments/assets/f4aaee61-abb8-4807-a4f0-a6966d20c232

now I don't have the issue

**Comment by remidej** [comment]
@simotae14 thanks! Your video is what we expect indeed. So maybe it was an issue with your environment, or maybe there's a tricky edge case here that we haven't identified. I'll merge for now and ask people to pay special attention to this during the next blitz.

**Pull request merged**